packages/perseus-core/src/parse-perseus-json/perseus-parsers/widgets-map.ts

```diff
@@ -211,14 +211,18 @@ const parseDeprecatedWidget: Parser<DeprecatedStandinWidget> = parseWidget(
     object({}),
 );
 
-const parseStringToPositiveInt: Parser<number> = (rawValue, ctx) => {
-    if (typeof rawValue !== "string" || !/^[1-9][0-9]*$/.test(rawValue)) {
+const parseStringToNonNegativeInt: Parser<number> = (rawValue, ctx) => {
+    // The article renderer seems to allow the numeric part of a widget ID to
+    // be 0, at least for image widgets. However, if widget IDs in an exercise
+    // contain 0, the exercise renderer will blow up. We allow 0 here for
+    // compatibility with articles.
+    if (typeof rawValue !== "string" || !/^(0|[1-9][0-9]*)$/.test(rawValue)) {
```

Comment:
```suggestion
    if (typeof rawValue !== "string" || !/^(\d+)$/.test(rawValue)) {
```

I'm guessing we can't do this because we don't want `01`?

I'm a little worried about codifying the idea that a widget ID is `widget 1` whereas where we're trying to get to is widget ID is something like `/[a-z0-9 -]+/` (a uuid) - it just happens that they are made of a widget type and a number now.

I know `parseStringToNonNegativeInt` is more generic than just an ID, but I thought I'd bring it up since the comment brings up widget IDs.

Reply:
> I'm guessing we can't do this because we don't want 01?

Yeah, a leading `0` causes the number to be parsed as octal, so e.g. `011` will parse to 9.

> I'm a little worried about codifying the idea that a widget ID is widget 1 whereas where we're trying to get to is widget ID is something like /[a-z0-9 -]+/ (a uuid) - it just happens that they are made of a widget type and a number now.

It's easy to relax constraints, but harder to tighten them (since we'd have to make sure that our existing data adheres to the tighter constraint). I think it will be straightforward to allow more widget ID formats in the future.